dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -431,6 +438,27 @@ def is_match(self) -> bool:
             is_match = True
         return is_match
 
+    def _preprocess_for_categorical_psi_calculation(
+        self, self_cat_count, other_cat_count
+    ):
+        super_set_categories = set(self_cat_count.keys()) | set(other_cat_count.keys())
+        if (super_set_categories != self_cat_count.keys()) or (
+            super_set_categories != other_cat_count.keys()
+        ):
+            logger.info(
+                f"""PSI data pre-processing found that categories between
+                    the profiles were not equal. Both profiles not contain
+                    the following categories {super_set_categories}."""
+            )
+
+        for iter_key in super_set_categories:
+            for iter_dictionary in [self_cat_count, other_cat_count]:
+                try:
+                    iter_dictionary[iter_key] = iter_dictionary[iter_key]
+                except KeyError:
+                    iter_dictionary[iter_key] = 0
+        return self_cat_count, other_cat_count
+
```

Comment:
main fix is here to ensure that each cat_count has the same keys even if some are zero.... this is to ensure the PSI is calculated when new categories are added or old categories are removed over time